modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetHandler.java

```diff
@@ -86,14 +89,27 @@
     public static final String MESSAGE_SIZE_VALIDATION = "message.size.validation.enabled";
     public static final String VALID_MAX_MESSAGE_SIZE = "valid.max.message.size.in.bytes";
 
+    private List<StreamInterceptor> streamInterceptors;
+    private boolean interceptStream;
+    private int noOfInterceptors;
+
+    public TargetHandler(DeliveryAgent deliveryAgent, ClientConnFactory connFactory,
+                         TargetConfiguration configuration) {
+        this(deliveryAgent, connFactory, configuration, new ArrayList<StreamInterceptor>());
+    }
+
     public TargetHandler(DeliveryAgent deliveryAgent,
                          ClientConnFactory connFactory,
-                         TargetConfiguration configuration) {
+                         TargetConfiguration configuration,
```

Comment:
Can this constructor is used by outside components? if so shall we use a new constructor with new parameters rather than doing an API change

Reply:
another constructor is defined already